core/ArchiveProcessor.php

```diff
@@ -668,4 +684,17 @@ public function getArchiveWriter()
     {
         return $this->archiveWriter;
     }
+
+    public function isArchiving(string $reportName)
+    {
+        $onlyReportName = $this->getParams()->getArchiveOnlyReport();
+        return empty($onlyReportName) || $onlyReportName == $reportName;
+    }
+
+    private function checkArchiveOnlyReport(string $name)
+    {
+        if ($name == $this->getParams()->getArchiveOnlyReport()) {
```

Comment:
not quite sure about the purpose behind this? and does it need to be reset in `this->getPArams()` again? eg after inserting a numeric record?

Reply:
It's to automatically check if we archived the one report that was requested, and if so, set the value to DONE_PARTIAL. You're right it needs to be reset if a different archive is inserted at any time. I guess it can't be done automatically in an easy way...